trie/triedb/pathdb/nodebuffer.go

```diff
@@ -37,6 +38,7 @@ type nodebuffer struct {
 	size   uint64                                    // The size of aggregated writes
 	limit  uint64                                    // The maximum memory allowance in bytes
 	nodes  map[common.Hash]map[string]*trienode.Node // The dirty node set, mapped by owner and path
+	mux    sync.RWMutex
```

Comment:
Locking because we support parallel execution?

Reply:
yes, the scenarios where read and write concurrency exist.